website/static/js/home-page/newAndNoteworthyPlugin.js

```diff
@@ -0,0 +1,166 @@
+            Raven.captureMessage('Error loading new and noteworthy projects on home page.', {requestReturn: result});
+        };
+
+        // Load new and noteworthy nodes
+        var newAndNoteworthyUrl = $osf.apiV2Url('nodes/' + window.contextVars.newAndNoteworthy + '/node_links/', {});
+        var newAndNoteworthyPromise = m.request({method: 'GET', url: newAndNoteworthyUrl, config: xhrconfig, background: true});
+        newAndNoteworthyPromise.then(function(result){
+            var numNew = Math.min(result.data.length, self.SHOW_TOTAL);
+            for (var l = 0; l < numNew; l++) {
+                self.newAndNoteworthyNodes().push(result.data[l]);
+                self.fetchContributors(result.data[l]);
+            }
+            self.someDataLoaded(true);
+        }, function _error(result){
+            self.requestError(result);
+        });
+
+        // Load popular nodes
+        var popularUrl = $osf.apiV2Url('nodes/' + window.contextVars.popular + '/node_links/', {});
+        var popularPromise = m.request({method: 'GET', url: popularUrl, config: xhrconfig, background: true});
+        popularPromise.then(function(result){
+            var numPopular = Math.min(result.data.length, self.SHOW_TOTAL);
+            for (var l = 0; l < numPopular; l++) {
+                self.popularNodes().push(result.data[l]);
+                self.fetchContributors(result.data[l]);
+            }
+            self.someDataLoaded(true);
+        }, function _error(result){
+            self.requestError(result);
+        });
+
+        // Additional API call to fetch node link contributors
+        self.fetchContributors = function(nodeLink) {
+            var url = nodeLink.embeds.target_node.data.relationships.contributors.links.related.href;
+            var promise = m.request({method: 'GET', url : url, config: xhrconfig});
+            promise.then(function(result){
+                var contribNames = [];
+                result.data.forEach(function (contrib){
+                    contribNames.push($osf.findContribName(contrib.embeds.users.data.attributes));
+                });
+                var numContrib = result.links.meta.total;
+                var nodeId = nodeLink.id;
+                self.contributorsMapping[nodeId] = [contribNames, numContrib];
+            }, function _error(result){
+                self.requestError(result);
+            });
+        };
+
+        // Gets contrib family name for display
+        self.getFamilyName = function(i, node) {
+            return self.contributorsMapping[node.id][0][i];
+        };
+
+        self.addToolTip = function(line) {
+            var $line = $(line);
+            if (line.offsetWidth < line.scrollWidth) {
+                $line.tooltip('show');
+            }
+        };
+    },
+    view : function(ctrl) {
+        if (ctrl.errorLoading()) {
+            return m('p.text-center.m-v-lg', 'Error loading projects. Please refresh the page.');
```

Comment:
For consistency with other error messages on the site, it would be good to include our support email address.